processing/src/main/java/org/apache/carbondata/processing/loading/constants/DataLoadProcessorConstants.java

```diff
@@ -38,4 +38,6 @@
 
   public static final String FACT_FILE_PATH = "FACT_FILE_PATH";
 
+  // to indicate that it is optimized insert flow without rearrange of each data rows
+  public static final String NO_REARRANGE_OF_ROWS = "NO_REARRANGE_OF_ROWS";
```

Comment:
please add a default value instead of giving true or false

Reply:
This is not a carbon property or user-configurable. All the places just checked whether this key is present. It doesn't care about value.